Make the switcher behave more like gnome workspaces do

AdirShemesh opened the pull request:
consider the case in the README, after calling next twice we're left with:
```
+-+-+-+-+
|1|2|3|5| 
+-+-+-+-+
```
but if we now call prev we get back to
```
+-+-+-+-+
|1|2|3|4| 
+-+-+-+-+
```
and only after another prev we get
```
+-+-+
|1|2|3|
+-+-+
```
Which gets annoying in my workflow since I end up having to switch through multiple empty workspaces to get to the desired one.

Thanks for the useful project :)

AdirShemesh committed:
Make the switcher behave more like gnome workspaces do

grenewode committed:
Multihead Support

Previously, the new changes meant that the switcher would "grab"
workspaces from other screens as you were switchering, rather then skipping
over them, which is the desired behavior for now.

For example:
A, B are screens
* workspace is not on a screen

Old behavior

screen   : A | B | *
workspace: 1 | 2 | 3

if we are on workspace 1 and say i3switcher next
then we get
screen   : B | A | *
workspace: 1 | 2 | 3

The new behavior is:
if we are on workspace 1 and say i3switcher next
then we get
screen   : * | B | A
workspace: 1 | 2 | 3

Note that in this case, workspace 2 stays on screen B

grenewode commented:
I really like this! It's been bugging me too. I just added a bit of stuff so that this will also work on setups with multiple screens. 

This seems to work on my setup, but please test it and report here if it works for you as well. Unless there are issues, or if I don't from you in a couple of days, I'll just go ahead and merge it.

Also, you're welcome. :smile: I'm just glad some is using it!

One minor note for future contributions: please run [rustfmt](https://github.com/rust-lang-nursery/rustfmt) on your commits! It helps keep them clean, without editor noise & such.

grenewode committed:
Better behavior for i3switcher prev

Previously, the tool would not create new workspaces when the user moved
to previous workspaces. This causes it to try to create a new workspace
if there is room for one in the numbered workspaces

AdirShemesh commented:
Your changes work on my machine too :+1:
